client/src/rubin/nublado/client/exceptions.py

```diff
@@ -369,12 +436,46 @@ def to_slack(self) -> SlackMessage:
 class JupyterWebError(NubladoClientSlackWebException):
     """An error occurred when talking to JupyterHub or a Jupyter lab."""
 
+    @classmethod
+    def raise_from_exception_with_started_at(
```

Comment:
`from_exception` didn't always take a `started_at`, although if we use `now` if we don't have one as a default it's probably OK?  I'll try that too.

Reply:
from_exception doesn't take started_at.  https://github.com/lsst-sqre/safir/blob/29f2532b3865cfca020f560e7da01df679b6cc14/safir/src/safir/slack/blockkit.py#L333

This is part of the overly-messy class hierarchy stuff.

Reply:
Gah, yep, I think I saw the `JupyterWebsocketError` `from_exception` later in the diff and thought it was this class.  What about overriding `from_exception` here instead, to stay consistent with the method naming?

Reply:
So, the SlackWebException derived ones, no, because those are expecting an HTTPError and changing the signature makes them unpickleable or something.

This thing is a complete nightmare, honestly.